crates/ursa-gateway/src/resolver/mod.rs

```diff
@@ -133,11 +134,23 @@ fn choose_provider(indexer_response: IndexerResponse) -> Result<Vec<SocketAddrV4
         .context("Indexer result did not contain a multi-hash result")?
         .provider_results;
 
-    if providers.is_empty() {
-        bail!("Multi-hash result did not contain a provider")
+    let provider = providers
+        .first()
+        .context("Multi-hash result did not contain a provider")?;
+
+    let metadata_bytes = base64::decode(&provider.metadata).map_err(|e| anyhow!("{e}"))?;
```

Comment:
If the context is not provided, there is a shorter sugar.
```suggestion
    let metadata_bytes = base64::decode(&provider.metadata)?;
```